ui/raidboss/data/05-shb/alliance/the_puppets_bunker.js

```diff
@@ -1,11 +1,547 @@
+        }
+
+        // Note: do not call out "avoid tank" here because there's a lot of markers going out.
+      },
+      run: (data) => delete data.busterTargets,
+    },
+    {
+      id: 'Puppet Compound 2P Three Parts Disdain',
+      netRegex: NetRegexes.headMarker({ id: '003E' }),
+      condition: (data) => data.phase === 'compound',
+      response: Responses.stackOn(),
+    },
+    {
+      id: 'Puppet Compound 2P Four Parts Resolve',
+      netRegex: NetRegexes.headMarker({ id: ['004F', '0050', '0051', '0052'] }),
+      condition: Conditions.targetIsYou(),
+      alertText: function(data, matches) {
+        return {
+          '004F': {
+            en: 'Jump #1 on YOU',
+          },
+          '0050': {
+            en: 'Cleave #1 on YOU',
+          },
+          '0051': {
+            en: 'Jump #2 on YOU',
+          },
+          '0052': {
+            en: 'Cleave #2 on YOU',
+          },
+        }[matches.id];
+      },
+    },
+    {
+      id: 'Puppet Compound 2P Energy Compression',
+      netRegex: NetRegexes.startsUsing({ source: 'Compound 2P', id: '51A6', capture: false }),
+      delaySeconds: 4,
+      infoText: {
+        en: 'Get Towers',
+        de: 'Türme nehmen',
+        fr: 'Prenez les tours',
+        ja: '塔を踏む',
+        ko: '장판 들어가기',
+        cn: '踩塔',
+      },
+    },
+    {
+      id: 'Puppet Compound Pod R011: Laser',
+      netRegex: NetRegexes.startsUsing({ source: 'Compound Pod', id: '541B', capture: false }),
+      suppressSeconds: 2,
+      // TODO: maybe this could be smarter and we could tell you where to go??
+      infoText: {
+        en: 'Avoid Lasers',
+      },
+    },
+    {
+      id: 'Puppet Puppet 2P Prime Blade Puppet Guaranteed In',
+      netRegex: NetRegexes.startsUsing({ source: 'Puppet 2P', id: '5421', capture: false }),
+      suppressSeconds: 2,
+      // TODO: have only seen this happen for the guaranteed Puppet In at 6250.7 with 4 clones.
+      // TODO: can this happen at other times??
+      alertText: {
+        en: 'Get Under Clone Corner',
+      },
+    },
+    {
+      id: 'Puppet Puppet 2P Prime Blade Puppet In',
+      netRegex: NetRegexes.startsUsing({ source: 'Puppet 2P', id: '519A', capture: false }),
+      suppressSeconds: 2,
+      // TODO: when I've seen this happen at 6379.4, it's been two clones, that start
+      // at corners and then teleport to two cardinals across from each other with fake
+      // teleports on the other cardinals.  Can this ability be corners instead??
+      alertText: {
+        en: 'Get Under Clone',
+      },
+    },
+    {
+      id: 'Puppet Puppet 2P Prime Blade Puppet Out Corner',
+      netRegex: NetRegexes.startsUsing({ source: 'Puppet 2P', id: '5198', capture: false }),
+      suppressSeconds: 2,
+      // TODO: have only seen this where the 4 clones stay out at 6379.4.
+      // TODO: have seen a report on a guide that the clones can teleport in??
+      alertText: {
+        en: 'Avoid Clones',
+      },
+    },
```

Comment:
I talked some with Panic about this, but I'm a little bit unsure about what the possibilities are with these puppets.  I think everything is guaranteed up until the 6379.4 block.  On that one, I have seen both 519A (with two clones ending on cardinals you need to get in) and 5198 (with four clones on corners where you need to hide middle).

https://gamerescape.com/2020/08/14/ffxiv-shadowbringers-guide-the-puppets-bunker/ suggests that maybe there's a case where you have to hide outside to dodge a "get out" sword attack?

I'm going to run this through test_timeline and see if we get any mismatches on 6379.4 and if there's another ability here (like the other "out" sword id).  Alternatively, I'd love to know if anybody sees "get in, but corners" for 6379.4.

Reply:
I ran test_timeline through 20+ timelines and it's clear that the only two abilities that the clones do are 5198 (out) and 519A (in).  However, I didn't quite do enough testing to figure out how many clones are doing that attack or if there's variations.

I think I'm going to stick with the original thought, that this variation is one of two things, either "get under cardinal clone" or "get middle", and we'll know when somebody complains that there's more possibilities.  I sure haven't seen them in 5-6 iterations of this.